sdk/src/sdk/pubsub/MqttPubSubClient.cpp

```diff
@@ -93,6 +109,36 @@ class MqttPubSubClient : public IPubSubClient, private mqtt::callback {
         m_client.publish(topic, data)->wait();
     }
 
+    PublishStatus publishOnTopic(const std::string& topic, const std::string& data,
+                                 int timeout_ms) override {
+        try {
```

Comment:
I know the existing function is block-waiting, but for this new function it could make sense to return a future or a AsyncResult to let the user decide if they like to do blocking wait or will use Callbacks for async notification (or likes to ignore outcome at all).

In case of keeping the blocking variant, you should think about using Paho mqtt's waitFor() ...

Reply:
Ok - accepted